gir/gir.go

```diff
@@ -114,6 +114,39 @@ type PkgRepository struct {
 	Path string // gir file path
 }
 
+// FindGIRFiles finds gir files from the given list of pkgs.
+func FindGIRFiles(pkgs ...string) ([]string, error) {
+	girDirs, err := pkgconfig.GIRDirs(pkgs...)
+	if err != nil {
+		return nil, err
+	}
+	visited := make(map[string]struct{}, len(girDirs))
+	var girFiles []string
+	// Iterate over `pkgs` instead of over `girDirs` directly, so
+	// that the order of `girFiles` is predictable based on order
+	// of the input arguments.  At least this was important to me
+	// for debugability, but I feel like I had another reason that
+	// I no longer remember.
+	for _, pkg := range pkgs {
+		girDir := girDirs[pkg]
+		if _, ok := visited[girDir]; ok {
```

Comment:
Is the only time this is useful is when `pkgs` contains multiple of the same entry? If so, would `pkgs = slices.Compact(pkgs)` work instead?

Reply:
Yeah, `visited` is just for when there are multiples of the same `girDir`. `pkgs = slices.Compact(pkgs)` doesn't quite get us there,
- it assumes that duplicates are grouped together, which isn't necessarily true (we could sort them first to make it true though)
- it assumes that each `pkg` has its own `girDir`, which isn't true (for example: `gtk4`, `gtk4-x11`, and `gtk4-wayland` all share `/nix/store/niw855nnjgqbq2s0iqxrk9xs5mr10rz8-gtk4-4.2.1-dev/share/gir-1.0`)

Reply:
These seem like good enough reasons to me.

Reply:
Could you add that as comments?